beacon-chain/sync/initial-sync/service.go

```diff
@@ -71,6 +71,28 @@ func WithVerifierWaiter(viw *verification.InitializerWaiter) Option {
 	}
 }
 
+// WithSyncChecker registers the initial sync service
+// in the checker.
+func WithSyncChecker(checker *SyncChecker) Option {
+	return func(service *Service) {
+		checker.Svc = service
+	}
+}
+
+// SyncChecker allows other services to check the current status of
+// initial-sync and use that internally in their service.
+type SyncChecker struct {
+	Svc *Service
+}
```

Comment:
The checker having the full service as a dependency doesn't look right to me... It seems like the initial sync service already implements the `SyncChecker` interface, you can use it as the blockchain's service dependency. One service depending on another is more natural.

Reply:
The main reason you cant do it is because initial sync has a dependency on the blockchain service. It is the reason the checker has been implemented